BluetoothAdvertisementsKotlin/app/src/main/java/com/example/bluetoothadvertisements/MainActivity.kt

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (C) 2021 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bluetoothadvertisements
+
+import android.bluetooth.BluetoothAdapter
+import android.bluetooth.BluetoothManager
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.view.Menu
+import android.view.MenuItem
+import androidx.appcompat.app.AppCompatActivity
+import com.example.bluetoothadvertisements.databinding.ActivityMainBinding
+
+private const val TAG = "MainActivity"
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityMainBinding
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        setSupportActionBar(findViewById(R.id.toolbar))
+
+        binding = ActivityMainBinding.inflate(layoutInflater)
+
+        if (savedInstanceState == null) {
+            verifyBluetooth()
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+
+        if (resultCode == RESULT_OK) {
+            when (requestCode) {
+                REQUEST_ENABLE_BT -> {
+                    Log.d(TAG, "onActivityResult: REQUEST_ENABLE_BT")
+                    verifyBluetooth()
+                }
+                PERMISSION_REQUEST_LOCATION -> {
+                    Log.d(TAG, "onActivityResult: PERMISSION_REQUEST_COARSE_LOCATION")
+                    verifyBluetooth()
```

Comment:
It's a little confusing that both switch statements point to verifyBluetooth()...

Can you comment what these are coming back from for each? I get the location must be a permission request... I didn't get (until reading much later) that the enable BT might be sending them to some settings page to enable it.

Also, can verifyBluetooth() be moved out or is there another request code case... maybe you should add an else that sends a message that it's an unrecognized request code.